experimental/origin-graphql/src/mutations/marketplace/_validation.js

```diff
@@ -0,0 +1,26 @@
+import contracts from '../../contracts'
+
+export async function validateNewOffer(listingId, data) {
+  const listing = await contracts.eventSource.getListing(listingId)
+
+  const unitsAvailable = Number(listing.unitsAvailable)
+  const offerQuantity = Number(data.quantity)
+  if (offerQuantity > unitsAvailable) {
+    throw new Error(`Insufficient units available (${unitsAvailable}) for offer (${offerQuantity})`)
+  }
+
+  const depositAvailable = web3.utils.toBN(listing.depositAvailable)
```

Comment:
Can we refer to contracts.web3 so we don't rely on a web3 global

Reply:
I've added a local `web3`